redbeams/src/main/java/com/sequenceiq/redbeams/filter/RequestIdFilter.java

```diff
@@ -0,0 +1,32 @@
+package com.sequenceiq.redbeams.filter;
+
+import com.sequenceiq.redbeams.service.ThreadBasedRequestIdProvider;
+
+import java.io.IOException;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ * A filter that sets the value of the request ID header in a thread-local.
+ */
+public class RequestIdFilter extends OncePerRequestFilter {
```

Comment:
Why do we need this class in redbeams? Why isn't it just a shared utility class?